File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/carpenterswall/domain/CarpentersWallObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.carpenterswall.domain;

public abstract class CarpentersWallObject {
    public abstract String objAsString();
    public boolean isHint() {return false;}
    public static CarpentersWallObject objTypeFromString(String str) {
        switch (str) {
            case "wall":
                return new CarpentersWallWallObject();
            case "marker":
                return new CarpentersWallMarkerObject();
            default:
                return new CarpentersWallEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/carpenterswall/CarpentersWallObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.carpenterswall

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

enum class CarpentersWallObject {
    Empty, Marker, Corner, Wall, Up, Down, Left, Right;
    val isHint get() = this == Corner || this == Up || this == Down || this == Left || this == Right
}

class CarpentersWallGameMove(val p: Position, var obj: CarpentersWallObject = CarpentersWallObject.Empty)
